wallet-kit/src/main/kotlin/bitcoin/wallet/kit/hdwallet/Mnemonic.kt

```diff
@@ -0,0 +1,133 @@
+package bitcoin.wallet.kit.hdwallet
+
+import WordList
+import bitcoin.wallet.kit.hdwallet.utils.EntropyGenerator
+import bitcoin.wallet.kit.hdwallet.utils.Hasher
+import bitcoin.wallet.kit.hdwallet.utils.PBKDF2SHA512
+import javax.inject.Inject
+import kotlin.experimental.and
+
+
+class Mnemonic @Inject constructor(private val entropyGenerator: EntropyGenerator) {
```

Comment:
Would be good to avoid injection here because the `EntropyGenerator` used only once when generating mnemonic keys and nowhere needed. For tests we can use powermock.